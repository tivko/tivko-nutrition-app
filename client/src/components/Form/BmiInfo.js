import { Text, View } from "react-native";
import React from "react";
import { colors, fontsSize } from "../../styles/base";

const BmiInfo = ({ totalRecomendations }) => {
  return (
    <View style={{ alignItems: "center" }}>
      <Text style={{ fontSize: fontsSize.sm, color: colors.text }}>
        ІМТ: {totalRecomendations?.bmi}
      </Text>
      <Text style={{ fontSize: fontsSize.sm, color: colors.text }}>
        Рекомендована вага:{" "}
        {totalRecomendations?.recommendedWeight?.lowerWeight} -{" "}
        {totalRecomendations?.recommendedWeight?.upperWeight} кг
      </Text>
    </View>
  );
};

export default BmiInfo;
