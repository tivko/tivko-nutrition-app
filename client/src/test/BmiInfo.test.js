import React from "react";
import renderer from "react-test-renderer";
import BmiInfo from "../components/Form/BmiInfo";

describe("<BmiInfo />", () => {
  it("renders correctly with given props", () => {
    const mockRecomendations = {
      bmi: 22,
      recommendedWeight: {
        lowerWeight: 50,
        upperWeight: 70,
      },
    };

    const tree = renderer
      .create(<BmiInfo totalRecomendations={mockRecomendations} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly without props", () => {
    const tree = renderer.create(<BmiInfo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
