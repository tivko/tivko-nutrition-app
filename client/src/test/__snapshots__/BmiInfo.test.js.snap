// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BmiInfo /> renders correctly with given props 1`] = `
<View
  style={
    {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#808080",
        "fontSize": 24,
      }
    }
  >
    ІМТ: 
    22
  </Text>
  <Text
    style={
      {
        "color": "#808080",
        "fontSize": 24,
      }
    }
  >
    Рекомендована вага:
     
    50
     -
     
    70
     кг
  </Text>
</View>
`;

exports[`<BmiInfo /> renders correctly without props 1`] = `
<View
  style={
    {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#808080",
        "fontSize": 24,
      }
    }
  >
    ІМТ: 
  </Text>
  <Text
    style={
      {
        "color": "#808080",
        "fontSize": 24,
      }
    }
  >
    Рекомендована вага:
     
     -
     
     кг
  </Text>
</View>
`;
